api/tests/Feature/NotificationTest.php

```diff
@@ -0,0 +1,467 @@
+        $this->assertNotNull($response->json('data.markAllNotificationsAsRead.1.readAt'));
+    }
+
+    public function testOnlyUnreadQuery()
+    {
+        $this->user->notify(
+            new PoolCandidateStatusChanged(
+                PoolCandidateStatus::SCREENED_OUT_NOT_INTERESTED->name,
+                PoolCandidateStatus::QUALIFIED_WITHDREW->name,
+                $this->pool->id,
+                $this->pool->name
+            )
+        );
+
+        $this->user->notify(
+            new PoolCandidateStatusChanged(
+                PoolCandidateStatus::NEW_APPLICATION->name,
+                PoolCandidateStatus::PLACED_CASUAL->name,
+                $this->pool->id,
+                $this->pool->name
+            )
+        );
+
+        $this->user->notifications()->first()->markAsRead();
+
+        $unreadNotifications = $this->user->unreadNotifications()->get()->toArray();
+        $response = $this->actingAs($this->user, 'api')
+            ->graphQL($this->queryNotifications, [
+                'where' => [
+                    'onlyUnread' => true,
+                ],
+            ]);
+
+        $responseNotifications = $response->json('data.notifications.data');
+        $this->assertCount(count($unreadNotifications), $responseNotifications);
+
+        foreach ($responseNotifications as $shouldBeUnread) {
+            $this->assertNull($shouldBeUnread['readAt']);
+        }
+    }
+
+    public function testDateRangeFilter()
+    {
+        $this->user->notify(
+            new PoolCandidateStatusChanged(
+                PoolCandidateStatus::PLACED_INDETERMINATE->name,
+                PoolCandidateStatus::QUALIFIED_AVAILABLE->name,
+                $this->pool->id,
+                $this->pool->name
+            )
+        );
+
+        $pastDate = new Carbon(config('constants.past_date'));
+        $this->user->notifications()->first()->update([
+            'read_at' => $pastDate,
+        ]);
+        $this->user->unreadNotifications->markAsRead();
+
+        $from = $pastDate->addDay();
+        $response = $this->actingAs($this->user, 'api')
+            ->graphQL($this->queryNotifications, [
+                'where' => [
+                    'readAt' => [
+                        'from' => $from->toDateTimeString(),
+                        'to' => Carbon::now()->toDateTimeString(),
+                    ],
+                ]]);
+
+        $notifications = $this->user->notifications()->get()->toArray();
+
+        $rangeResponse = $response->json('data.notifications.data');
+
+        $this->assertCount(count($notifications) - 1, $rangeResponse);
+
+        foreach ($rangeResponse as $shouldBeInRange) {
+            $date = new Carbon($shouldBeInRange['readAt']);
+
+            $this->assertTrue($date->gt($from));
+
+        }
+    }
+
+    /**
+     * Test that notifications are generated for status changes
+     *
+     * @return void
+     */
+    public function testStatusChangeCausesNotifications()
```

Comment:
Sorry, after taking a closer look at this, I think your other suggestion was correct. This deserves to be in its own file, specific to the Status Change notification.

Reply:
Oh yeah, that one seems a little too specific for this file. Moved out in [reorganized notification tests](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9694/commits/179699d422207e642706fd900a0538072d03a175)